From: Casey Bodley <cbodley@redhat.com>
Date: Thu, 1 Dec 2016 13:53:44 -0500
Subject: rgw: fix for versioned delete_multi_object

Fixes: http://tracker.ceph.com/issues/17888

Signed-off-by: Casey Bodley <cbodley@redhat.com>
(cherry picked from commit cc2d1e6100081aaca300ade51617294a37d33b9f)
(cherry picked from commit 820ab7d8cf57a4cb4d8a550d3f8b5856026fc354)

Resolves (with previous): rhbz#1399624
---
 src/rgw/rgw_op.cc | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/rgw/rgw_op.cc b/src/rgw/rgw_op.cc
index 79e244a..cdff547 100644
--- a/src/rgw/rgw_op.cc
+++ b/src/rgw/rgw_op.cc
@@ -4449,8 +4449,8 @@ void RGWDeleteMultiObj::execute()
   for (iter = multi_delete->objects.begin();
         iter != multi_delete->objects.end() && num_processed < max_to_delete;
         ++iter, num_processed++) {
-    rgw_obj obj(bucket, (*iter).name);
-    obj.set_instance(s->object.instance);
+    rgw_obj obj(bucket, iter->name);
+    obj.set_instance(iter->instance);
 
     obj_ctx->set_atomic(obj);
 
